packages/flutter/lib/src/material/button.dart

```diff
@@ -421,3 +445,34 @@ class MaterialButton extends StatelessWidget {
     properties.add(new FlagProperty('enabled', value: enabled, ifFalse: 'disabled'));
   }
 }
+
+/// Redirects the position passed to [RenderBox.hitTest] to the center of the widget.
+///
+/// The primary purpose of this widget is to allow padding around [Material] widgets
+/// to trigger the child ink feature without increasing the size of the material.
+class _RedirectingHitDetectionWidget extends SingleChildRenderObjectWidget {
+  const _RedirectingHitDetectionWidget({this.constraints, Widget child}) : super(child: child);
+
+  final BoxConstraints constraints;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return _RedirectingHitDetectionRenderObject(constraints);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RedirectingHitDetectionRenderObject renderObject) {
+    renderObject.additionalConstraints = constraints;
+  }
+}
+
+class _RedirectingHitDetectionRenderObject extends RenderConstrainedBox {
+  _RedirectingHitDetectionRenderObject(BoxConstraints additionalConstraints) : super(additionalConstraints: additionalConstraints);
+
+  @override
+  bool hitTest(HitTestResult result, {Offset position}) {
+    if (!size.contains(position))
+      return false;
+    return child.hitTest(result, position: size.center(Offset.zero));
+  }
+}
```

Comment:
this is an interesting approach.

Why can't we just make the single GestureDetector bigger though?

Reply:
The GestureDetector/InkWell/Material is a delicate ecosystem.

InkWell  (which contains the gesture detector) uses context to find a parent material where it draws the splashes.  Any increase in the size of the gesture detector, which is sized by the material, necessarily increase the material size which would increase the visible appearance of the button.